### ISSUE TITLE
NullPointerException com.hazelcast.internal.networking.nio.NioOutboundPipeline.executePipeline(NioOutboundPipeline.java:244)

### ISSUE THREAD

**Issue by Danny-Hazelcast** [hazelcast/hazelcast#19286]
hit very rare NPE on client site version 4.0

http://jenkins.hazelcast.com/view/Release/job/wan-5-star/130/console
/disk1/workspace/wan-5-star/5.0-BETA-2-SNAPSHOT/2021_08_08-11_02_09/5-star

```
INFO: hz.client_1 [A] [4.0] HazelcastClient 4.0 (20210803 - 0f63740, 8b959f5) is CLIENT_CONNECTED
Aug 08, 2021 12:18:43 PM com.hazelcast.client.impl.connection.ClientConnectionManager
INFO: hz.client_1 [A] [4.0] Authenticated with server [10.0.0.195]:5701:bf8c53f4-d5a0-4ed8-bea4-2ca3a071f4ae, server version: 4.0, local address: /10.0.0.111:38925
Aug 08, 2021 12:18:43 PM com.hazelcast.client.impl.spi.ClientClusterService
INFO: hz.client_1 [A] [4.0]

Members [2] {
	Member [10.0.0.195]:5701 - bf8c53f4-d5a0-4ed8-bea4-2ca3a071f4ae
	Member [10.0.0.93]:5701 - a734c19c-1489-4772-864e-d1c5d7b94b5b
}

Aug 08, 2021 12:18:43 PM com.hazelcast.client.impl.connection.ClientConnectionManager
INFO: hz.client_1 [A] [4.0] Authenticated with server [10.0.0.93]:5701:a734c19c-1489-4772-864e-d1c5d7b94b5b, server version: 4.0, local address: /10.0.0.111:50797
Aug 08, 2021 12:18:43 PM com.hazelcast.client.impl.connection.ClientConnectionManager
INFO: hz.client_1 [A] [4.0] Authenticated with server [10.0.0.111]:5701:c85569ab-c84e-4d14-ba1a-f2660ef67e69, server version: 4.0, local address: /10.0.0.111:43725
java.lang.NullPointerException
	at com.hazelcast.internal.networking.nio.NioOutboundPipeline.executePipeline(NioOutboundPipeline.java:244)
	at com.hazelcast.internal.networking.nio.NioOutboundPipeline.write(NioOutboundPipeline.java:215)
	at com.hazelcast.internal.networking.nio.NioChannel.write(NioChannel.java:79)
	at com.hazelcast.internal.nio.tcp.TcpIpConnection.write(TcpIpConnection.java:185)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.sendClientMessage(AbstractMessageTask.java:251)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.sendResponse(AbstractMessageTask.java:239)
	at com.hazelcast.client.impl.protocol.task.AbstractAsyncMessageTask.sendResponseOrHandleFailure(AbstractAsyncMessageTask.java:79)
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478)
	at com.hazelcast.internal.util.ConcurrencyUtil$1.execute(ConcurrencyUtil.java:39)
	at java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:886)
	at java.util.concurrent.CompletableFuture.whenCompleteAsync(CompletableFuture.java:2261)
	at com.hazelcast.client.impl.protocol.task.AbstractAsyncMessageTask.processMessage(AbstractAsyncMessageTask.java:73)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.initializeAndProcessMessage(AbstractMessageTask.java:145)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.run(AbstractMessageTask.java:108)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:181)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:172)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:140)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:123)
java.lang.NullPointerException
	at com.hazelcast.internal.networking.nio.NioOutboundPipeline.executePipeline(NioOutboundPipeline.java:244)
	at com.hazelcast.internal.networking.nio.NioOutboundPipeline.write(NioOutboundPipeline.java:215)
	at com.hazelcast.internal.networking.nio.NioChannel.write(NioChannel.java:79)
	at com.hazelcast.internal.nio.tcp.TcpIpConnection.write(TcpIpConnection.java:185)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.sendClientMessage(AbstractMessageTask.java:251)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.sendResponse(AbstractMessageTask.java:239)
	at com.hazelcast.client.impl.protocol.task.AbstractAsyncMessageTask.sendResponseOrHandleFailure(AbstractAsyncMessageTask.java:79)
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478)
	at com.hazelcast.internal.util.ConcurrencyUtil$1.execute(ConcurrencyUtil.java:39)
	at java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:886)
	at java.util.concurrent.CompletableFuture.whenCompleteAsync(CompletableFuture.java:2261)
	at com.hazelcast.client.impl.protocol.task.AbstractAsyncMessageTask.processMessage(AbstractAsyncMessageTask.java:73)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.initializeAndProcessMessage(AbstractMessageTask.java:145)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.run(AbstractMessageTask.java:108)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:181)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:172)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:140)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:123)
Aug 08, 2021 12:19:13 PM com.hazelcast.client.impl.connection.nio.ClientConnection
INFO: hz.client_1 [A] [4.0] ClientConnection{alive=false, connectionId=45, channel=NioChannel{/10.0.0.111:43725->/10.0.0.111:5701}, remoteEndpoint=[10.0.0.111]:5701, lastReadTime=2021-08-08 12:19:13.512, lastWriteTime=2021-08-08 12:19:12.727, closedTime=2021-08-08 12:19:13.512, connected server version=4.0} closed. Reason: Connection closed by the other side
Aug 08, 2021 12:19:13 PM com.hazelcast.client.impl.connection.ClientConnectionManager
INFO: hz.client_1 [A] [4.0] Removed connection to endpoint: [10.0.0.111]:5701:c85569ab-c84e-4d14-ba1a-f2660ef67e69, connection: ClientConnection{alive=false, connectionId=45, channel=NioChannel{/10.0.0.111:43725->/10.0.0.111:5701}, remoteEndpoint=[10.0.0.111]:5701, lastReadTime=2021-08-08 12:19:13.512, lastWriteTime=2021-08-08 12:19:12.727, closedTime=2021-08-08 12:19:13.512, connected server version=4.0}
Aug 08, 2021 12:19:19 PM com.hazelcast.client.impl.spi.ClientClusterService
INFO: hz.client_1 [A] [4.0]

Members [3] {
	Member [10.0.0.195]:5701 - bf8c53f4-d5a0-4ed8-bea4-2ca3a071f4ae
	Member [10.0.0.93]:5701 - a734c19c-1489-4772-864e-d1c5d7b94b5b
	Member [10.0.0.111]:5701 - ce23ae9b-d473-4b65-bba3-c9af1dcafa65
}
```

**Comment by Danny-Hazelcast** [comment]
we had this this issue  int 4.2.
 
https://github.com/hazelcast/hazelcast/issues/16754

**Comment by mmedenjak** [comment]
I guess one of these two need to be backported: https://github.com/hazelcast/hazelcast/pull/18363, https://github.com/hazelcast/hazelcast/pull/18364

**Comment by kwart** [comment]
Fixed by #19248
